### PR TITLE
Organization: add PROCESS.md and ROADMAP.md

### PR DESCRIPTION
**Pull request by laurentsenta**
- [ ] Who should review and be part of the discussion?
- [x] Add content from [https://github.com/libp2p/interop/](https://github.com/libp2p/interop/actions/runs/3021456724)
- [x] Make the tables readable in text view
- [x] Drop authors and reviewers section, that's handled by git(hub)

### CODE REVIEW

**Comment by BigLep** [comment]
Thanks for starting this @laurentsenta .  A few things:
1. What do we think it will take to merge this?  Basically, when are you considering it done?  Are you wanting to see all of these questions answered?  Just some of them?
2. An important part of this experience is when a test fails, is it pretty straightforward for someone to see who the culprit is?  For the "big test", does someone get a dump of all the combinations that were run and which ones passed/fail so they can identify the failure source (see notes on that below)?
3. I like the idea of having a dashboard that shows that, for example, the "Simple PING" "big test" here was the last run, here were all the implementations/versions that were run, and here was the result state (pass or fail) and execution time.  
4. Lets answer what happens in the "big tests".  For example, if I have these libp2p nodes go-vA.0, go-vB.0, rust-vC.0, rust-vD.0, is there a pairwise set of asserts that are run like, something like this:
```
   go-vA.0 / go-vB.0
   go-vA.0 / rust-vC.0
   go-vA.0 / rust-vD.0
   go-vB.0 / rust-vC.0
   go-vB.0 / rust-vD.0
   rust-vC.0 / rust-vD.0
```
?

Basically, what I'm wondering in a "big test" whether we'll be able to see which combinations ran, did they success, how long they took, etc.
Also, ideally that does run on a cron job and someone can come to libp2p/test-plans and click into the latest results.

**Comment by laurentsenta** [comment]
@BigLep 

> 1. What do we think it will take to merge this?  Basically, when are you considering it done?  Are you wanting to see all of these questions answered?  Just some of them?

- Ideally I'd like to import info about the `libp2p/interop` repository to mark this as ready for review,
- Then have the rest of the team review & share input,
- To me, every question below `When do we revisit this scenario to improve and gather feedback?` needs an answer. We probably don't have enough information to answer the other questions.

> 2. An important part of this experience is when a test fails, is it pretty straightforward for someone to see who the culprit is?  For the "big test", does someone get a dump of all the combinations that were run and which ones passed/fail so they can identify the failure source (see notes on that below)?
> 3. I like the idea of having a dashboard that shows that, for example, the "Simple PING" "big test" here was the last run, here were all the implementations/versions that were run, and here was the result state (pass or fail) and execution time.
> 4. Lets answer what happens in the "big tests".  For example, if I have these libp2p nodes go-vA.0, go-vB.0, rust-vC.0, rust-vD.0, is there a pairwise set of asserts that are run like, something like this:
> 
> ```
>    go-vA.0 / go-vB.0
>    go-vA.0 / rust-vC.0
>    go-vA.0 / rust-vD.0
>    go-vB.0 / rust-vC.0
>    go-vB.0 / rust-vD.0
>    rust-vC.0 / rust-vD.0
> ```
> 
> ?
> 
> Basically, what I'm wondering in a "big test" whether we'll be able to see which combinations ran, did they success, how long they took, etc. Also, ideally that does run on a cron job and someone can come to libp2p/test-plans and click into the latest results.

- We worked the other way around: implementations run tests to prevent merging "broken" code, they have to run quickly enough and signal that something is broken at best (not much more feedback).
- That approach makes a lot of sense and can be done easily for libp2p days 👍

**Comment by galargh** [comment]
1:1 notes:
- this is a high-priority task for libp2p day
- I (@galargh) can help out with this task to free up @laurentsenta (as he will have other tasks that have to be completed for libp2p day)
- we're going to meet up to discuss specific action items

File: ROADMAP.md
Comment (BigLep) [inline ROADMAP.md:22]: I know what these are, but they are at a different level than browserjs/nodejs.

It seems like we're mixing language implementations with transports.

I do think we need to get to a place of libp2p testing across various dimensions:

<language, transport, version>

Is enumerating transports (TCP, Quic, WebTransport, WebRTC) in scope for this milestone?

I guess what's happening here is that in order to show that browser/nodejs-libp2p can talk with go-libp2p or rust-libp2p, they need to use WebTransport and/or WebRTC.

To drive clarity, I think mocking out what the page looks like will help make sure everyone is on the same page.

File: ROADMAP.md
Comment (BigLep) [inline ROADMAP.md:23]: Great - thanks!

File: ROADMAP.md
Comment (BigLep) [inline ROADMAP.md:41]:
```suggestion
    - The design is documented in `./DESIGN.md`,
```

File: ROADMAP.md
Comment (BigLep) [inline ROADMAP.md:56]: There are dependencies on the testground roadmap for these.  It would be great to model this.  This is where bumping up to GitHub issues rather than markdown files can help us so that have something to link to.

File: DESIGN.md
Comment (BigLep) [inline DESIGN.md:32]: For webtransport, only Go is possible to test now (as only one that has the transport implemented).  JS is the only other implementation in-process on adding the transport.

```suggestion
| WebTransport Transport            | 🔥        |           | 🔥 (depends on https://github.com/libp2p/js-libp2p-webtransport/issues/1)               | 🔥 (depends on https://github.com/libp2p/js-libp2p-webtransport/issues/1)                  |          |          |
```

File: DESIGN.md
Comment (BigLep) [inline DESIGN.md:33]:
```suggestion
| WebRTC Transport                  | 🔥 (depends on working implementation)        | 🔥 (depends on working implementation)          | 🔥 (depends on working implementation)               | 🔥 (depends on working implementation)                  |          |          |
```

File: DESIGN.md
Comment (BigLep) [inline DESIGN.md:44]: Is there a tracking issue for this?

File: DESIGN.md
Comment (BigLep) [inline DESIGN.md:49]: Combine with your other "Questions" section below?

File: DESIGN.md
Comment (BigLep) [inline DESIGN.md:63]:
```suggestion
## Background
The approach outlined below is pretty much what happen with the go|rust-libp2p ping tests in 2022Q3.

libp2p implementations aren't forced to adopt this approach, but it is the approach that has been taken by some of the longer-lived implementations (go, JS, and rust).  

I (@laurent) haven’t had time to look at [libp2p/interop](https://github.com/libp2p/interop/actions/runs/3021456724) yet. Some information may be missing.
```

File: DESIGN.md
Comment (BigLep) [inline DESIGN.md:66]:
```suggestion
## 202210 Proposal
<aside>
1️⃣ Before working on a new feature, the libp2p maintainers come together and agree on a description of the new test plan.*
```
Emoji with bullet didn't render well.

File: DESIGN.md
Comment (BigLep) [inline DESIGN.md:123]:
```suggestion
## Open Questions
```

File: DESIGN.md
Comment (BigLep) [inline DESIGN.md:59]:
```suggestion
# How does libp2p test interoperability?
```

I want to move libp2p from being more than a PL project.

**Review by BigLep — COMMENTED** [review]
Thanks for starting this @laurentsenta and adding the ROADMAP.  I think you're first "short term" milestone is a great place to get libp2p/test-plans and let libp2p to take over from there.

If there is contention on the DESIGN.md file, I think it would also work for it to be in its own PR.

File: ROADMAP.md
Comment (mxinden) [inline ROADMAP.md:14]: Note, in case these items are ordered by priority:

In my eyes, for the libp2p team, covering essential features (item 2) is more important than a page to discover the results (item 1). In other words, I think we should first create the things before figuring out a way to display them.

**Review by mxinden — COMMENTED** [review]
Thank you @laurentsenta for writing this up. I am sorry I haven't taken a look earlier.

File: ROADMAP.md
Comment (laurentsenta) [reply ROADMAP.md:22]: Thanks for the feedback, just a quick note,

We created an epic for implementing test matrixes at https://github.com/libp2p/test-plans/issues/53
At the moment I am working on https://github.com/libp2p/test-plans/issues/55 which is step 1.

**Comment by p-shahi** [comment]
Marking this as ready to review with the aim of merging it today. It has gone through review in https://github.com/libp2p/test-plans/pull/56 as well as earlier here.

The next steps before merging are:

- [ ] address open comments

- [ ] add links to GitHub issues

The immediate projects have been assigned completion targets but some are missing them. These will be addressed in a followup as they need more investigation.

**Review by BigLep — APPROVED** [review]

File: ROADMAP.md
Comment (p-shahi) [reply ROADMAP.md]: Done

**Pull request merged**
